internal/state/extended_state.go

```diff
@@ -0,0 +1,14 @@
+package state
+
+// ExtendedState represents an extension to the default state object.
+type ExtendedState interface {
```

Comment:
I am bit confused by the naming. Is there a difference between "extended" and "external"? If I understand it right then if someone is using an extended state, from his point of view this is always external. 

So should this better be named `ExternalState` instead? 
Another thing we might think about is just calling it `External` as the package name already includes `state` but this might collide with the parent which has to be named `state.State`.